torch/csrc/Storage.cpp

```diff
@@ -51,6 +51,66 @@ static void THPStorage_subclass_dealloc(PyObject* self) {
   Py_TYPE(_self)->tp_free(self);
 }
 
+c10::intrusive_ptr<c10::StorageImpl> make_storage_impl(
+    c10::StorageImpl::use_byte_size_t use_byte_size,
+    c10::SymInt size_bytes,
+    c10::Allocator* allocator,
+    bool resizable,
+    c10::optional<int64_t> allocator_opt,
+    c10::optional<at::Device> device_opt) {
+  at::OptionalDeviceGuard device_guard;
+  c10::StorageImplCreateHelper fptr = nullptr;
```

Comment:
```suggestion
  // This will be non-nullptr only when there is a custom StorageImpl constructor for the given device
  c10::StorageImplCreateHelper fptr = nullptr;
```